### ISSUE TITLE
drivers: can: net: compilation broken and no test cases in CI

### ISSUE THREAD

**Issue by henrikbrixandersen** [zephyrproject-rtos/zephyr#40169]
**Describe the bug**
The `drivers/can/can_net.c` implementation is broken (does not compile, see below) and there are no in-tree build tests.
I wonder if anybody even uses this code? If not, one option could be to retire it and all that depends on it.

**To Reproduce**
1. `cd samples/net/sockets/echo_server`
2. `mkdir build; cd build`
3. `cmake -DBOARD=nucleo_f767zi -DOVERLAY_CONFIG=overlay-6locan.conf ..`
4. `make`
5. See error:

```
[ 62%] Building C object zephyr/drivers/can/CMakeFiles/drivers__can.dir/can_net.c.obj
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c: In function 'mcast_cb':
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c:191:44: error: incompatible type for argument 2 of 'attach_mcast_filter'
  191 |   filter_id = attach_mcast_filter(ctx, addr->in6_addr);
      |                                        ~~~~^~~~~~~~~~
      |                                            |
      |                                            const struct in6_addr
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c:144:34: note: expected 'const struct in6_addr *' but argument is of type 'const struct in6_addr'
  144 |           const struct in6_addr *addr)
      |           ~~~~~~~~~~~~~~~~~~~~~~~^~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c:197:26: error: incompatible types when assigning to type 'const struct in6_addr *' from type 'const struct in6_addr'
  197 |   filter_mapping->addr = addr->in6_addr;
      |                          ^~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c:200:50: error: incompatible type for argument 2 of 'can_get_mcast_filter'
  200 |   filter_mapping = can_get_mcast_filter(ctx, addr->in6_addr);
      |                                              ~~~~^~~~~~~~~~
      |                                                  |
      |                                                  const struct in6_addr
/home/brix/Projects/zephyrproject/zephyr/drivers/can/can_net.c:32:35: note: expected 'const struct in6_addr *' but argument is of type 'const struct in6_addr'
   32 |            const struct in6_addr *addr)
      |            ~~~~~~~~~~~~~~~~~~~~~~~^~~~
make[2]: *** [zephyr/drivers/can/CMakeFiles/drivers__can.dir/build.make:118: zephyr/drivers/can/CMakeFiles/drivers__can.dir/can_net.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:3777: zephyr/drivers/can/CMakeFiles/drivers__can.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

```

**Expected behavior**
Code compiles and works. Compilation to be tested in CI.

**Impact**
Presumably dead or unused code, adding to the complexity of the CAN drivers and stack.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: 9bea59a8c4b70ebd6da5fe2cd1566eb93ebbbdcf

**Comment by henrikbrixandersen** [comment]
@karstenkoenig, @nixward, @martinjaeger, @legoabram: FYI

**Comment by henrikbrixandersen** [comment]
Compilation and CI issue fixed in #40911.
